sdk/azcore/transport_default_http_client.go

```diff
@@ -31,9 +31,7 @@ func init() {
 	}
 }
 
-// DefaultHTTPClientTransport ...
+// DefaultHTTPClientTransport returns a shared Transport. It uses an http.Client.
```

Comment:
Should this doc string be a little more like our other default func doc strings? Something like: "...returns a shared Transport initialized with default values in an http.Client."

Reply:
You bring up a good point, and this makes me think that perhaps we should ditch the word "Default" altogether since it's different from `DefaultFooOptions()`.  Name it simply `HTTPClientTransport()` (no "New" prefix since this is shared).  What do you think?

Reply:
Hmmm...well I'm fine with default. But if you think this could cause customers confusion and they would want to change this then HTTPClientTransport would be ok. I do think Default describes this well. Maybe an addition needs to be made to the comment about when this is meant to be used? Or why this exists?

Reply:
If the customer doesn't specify a transport, then this IS the default transport, right?
I think this IS similar to our client options. In fact, we might want to add a TransportPolicyOptions struct and put the Transport as a field inside. Then we can get the DefaultTransportPolicyOptions and return the initialized structure with the field initialized to what this function returns. Then, we could actually delete this function.

This has 2 benefits: consistency with other policies and the ability to add fields to the transport policy without breaking in the future.

Reply:
Correct, this is the default if you pass `nil` as the transport arg in `NewPipeline()`.

I feel like we should simply delete `DefaultHTTPClientTransport()`, so if you want the default behavior pass `nil`.  If you want a custom `Transport`, pass in something that implements that interface.

Reply:
OK

Reply:
@catalinaperalta are you ok with this?

Reply:
Yes that sounds good to me 👍

Reply:
Fixed